BookShelfBL/Library.cs

```diff
@@ -0,0 +1,301 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace BookShelfBL
+{
+    /// <summary>
+    /// Class containing all methods for data exchange with the database.
+    /// Creates a temporary set of data to work with.
+    /// </summary>
+    public class Library : ILibrary
+    {
+        private List<Book> _books;
+        private List<Author> _authors;
+        private List<Genre> _genres;
+
+        /// <summary>
+        /// Creates new instance of Library, seeding some sample data
+        /// </summary>
+        public Library()
+        {
+            _authors = new List<Author>();
+            _authors.Add(new Author("Leo Tolstoy"));
+            _authors.Add(new Author("Marcel Proust")); 
+            _authors.Add(new Author("James Joyce"));
+            _authors.Add(new Author("William Shakespeare"));
+
+            this._books = new List<Book>();
+            _books.Add(new Book("War and Peace"));
+            _books.Add(new Book("Anna Korenina"));
+            _books.Add(new Book("Hamlet"));
+            _books.Add(new Book("The Odyssey"));
+            _books.Add(new Book("Ulysses"));
+            _books.Add(new Book("In Search of Lost Time"));
+            _books.Add(new Book("The Iliad"));
+
+            this._genres = new List<Genre>();
+            _genres.Add(new Genre("Classic"));
+            _genres.Add(new Genre("Modern"));
+            _genres.Add(new Genre("Ancient"));
+        }
+
+        // #####################################
+        // ALL METHODS RELATED TO BOOK:
+        // #####################################
+        #region BOOKS
+
+        /// <summary>
+        /// Gets the collection of all Books available in the data source
+        /// </summary>
+        /// <returns>The collection of Books </returns>
+        public List<Book> GetBooks()
+        {
+            return _books;
+        }
+
+        /// <summary>
+        /// Adds new book to data source
+        /// </summary>
+        /// <param name="book">New Book to be added</param>
+        /// <returns>Id of added book</returns>
+        public int AddBook(Book book)
+        {
+            _books.Add(book);
+
+            return book.Id;
+        }
+
+        /// <summary>
+        /// Gets the correct Book instance from the collection
+        /// </summary>
+        /// <param name="id"> Id of book to look for</param>
+        /// <returns> Book instance with specified Id or NULL </returns>
+        public Book GetBookById(int id)
+        {
+            return _books.Find(b => b.Id == id);
```

Comment:
FYI If id wouldn't find 'FIND' method will throw an exception Are you handling that case and do you ready for that and is it okay according to your logic?